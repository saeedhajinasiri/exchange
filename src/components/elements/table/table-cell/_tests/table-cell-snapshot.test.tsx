import React from "react";
import {cleanup, render} from "test";
import TableCell from '../table-cell'
import Table from '../../table'
import TableBody from '../../table-body'
import TableRow from '../../table-row'

afterEach(cleanup)

describe('Test TableCell UI', () => {
  it('Default', () => {
    const {asFragment} = render(<Table><TableBody><TableRow><TableCell>children</TableCell></TableRow></TableBody></Table>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('Test with children prop', () => {
    const {asFragment} = render(<Table><TableBody><TableRow><TableCell>children</TableCell></TableRow></TableBody></Table>);
    expect(asFragment()).toMatchSnapshot();
  });
});

describe('Test TableCell Variants', () => {
  it('variant="head', () => {
    const {asFragment} = render(<Table><TableBody><TableRow><TableCell variant={"head"}>children</TableCell></TableRow></TableBody></Table>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('variant="footer', () => {
    const {asFragment} = render(<Table><TableBody><TableRow><TableCell variant={"footer"}>children</TableCell></TableRow></TableBody></Table>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('variant="body', () => {
    const {asFragment} = render(<Table><TableBody><TableRow><TableCell variant={"body"}>children</TableCell></TableRow></TableBody></Table>);
    expect(asFragment()).toMatchSnapshot();
  });
});

describe('Test TableCell Sizes', () => {
  it('size=small', () => {
    const {asFragment} = render(<Table><TableBody><TableRow><TableCell size={"small"}>children</TableCell></TableRow></TableBody></Table>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('size=medium', () => {
    const {asFragment} = render(<Table><TableBody><TableRow><TableCell size={"medium"}>children</TableCell></TableRow></TableBody></Table>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('size=large', () => {
    const {asFragment} = render(<Table><TableBody><TableRow><TableCell size={"large"}>children</TableCell></TableRow></TableBody></Table>);
    expect(asFragment()).toMatchSnapshot();
  });
});

describe('Test TableCell Colspan', () => {
  it('colspan=2', () => {
    const {asFragment} = render(<Table><TableBody><TableRow><TableCell colSpan={2}>children</TableCell></TableRow></TableBody></Table>);
    expect(asFragment()).toMatchSnapshot();
  });
});
