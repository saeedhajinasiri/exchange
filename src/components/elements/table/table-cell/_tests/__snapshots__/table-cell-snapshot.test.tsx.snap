// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test TableCell Colspan colspan=2 1`] = `
<DocumentFragment>
  <table
    class="w-full"
  >
    <tbody
      class="table-row-group"
    >
      <tr
        class="border-b border-grey-100 border-opacity-20 hover:bg-opacity-75 bg-white border-grey-light"
      >
        <td
          class="px-4 bg-opacity-20 min-w-[125px] h-10"
          colspan="2"
        >
          <p
            class="text-m-sm sm:text-t-sm md:text-d-sm text-grey text-left font-regular"
          >
            children
          </p>
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;

exports[`Test TableCell Sizes size=large 1`] = `
<DocumentFragment>
  <table
    class="w-full"
  >
    <tbody
      class="table-row-group"
    >
      <tr
        class="border-b border-grey-100 border-opacity-20 hover:bg-opacity-75 bg-white border-grey-light"
      >
        <td
          class="px-4 bg-opacity-20 min-w-[125px] h-20"
        >
          <p
            class="text-m-sm sm:text-t-sm md:text-d-sm text-grey text-left font-regular"
          >
            children
          </p>
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;

exports[`Test TableCell Sizes size=medium 1`] = `
<DocumentFragment>
  <table
    class="w-full"
  >
    <tbody
      class="table-row-group"
    >
      <tr
        class="border-b border-grey-100 border-opacity-20 hover:bg-opacity-75 bg-white border-grey-light"
      >
        <td
          class="px-4 bg-opacity-20 min-w-[125px] h-15"
        >
          <p
            class="text-m-sm sm:text-t-sm md:text-d-sm text-grey text-left font-regular"
          >
            children
          </p>
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;

exports[`Test TableCell Sizes size=small 1`] = `
<DocumentFragment>
  <table
    class="w-full"
  >
    <tbody
      class="table-row-group"
    >
      <tr
        class="border-b border-grey-100 border-opacity-20 hover:bg-opacity-75 bg-white border-grey-light"
      >
        <td
          class="px-4 bg-opacity-20 min-w-[125px] h-10"
        >
          <p
            class="text-m-sm sm:text-t-sm md:text-d-sm text-grey text-left font-regular"
          >
            children
          </p>
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;

exports[`Test TableCell UI Default 1`] = `
<DocumentFragment>
  <table
    class="w-full"
  >
    <tbody
      class="table-row-group"
    >
      <tr
        class="border-b border-grey-100 border-opacity-20 hover:bg-opacity-75 bg-white border-grey-light"
      >
        <td
          class="px-4 bg-opacity-20 min-w-[125px] h-10"
        >
          <p
            class="text-m-sm sm:text-t-sm md:text-d-sm text-grey text-left font-regular"
          >
            children
          </p>
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;

exports[`Test TableCell UI Test with children prop 1`] = `
<DocumentFragment>
  <table
    class="w-full"
  >
    <tbody
      class="table-row-group"
    >
      <tr
        class="border-b border-grey-100 border-opacity-20 hover:bg-opacity-75 bg-white border-grey-light"
      >
        <td
          class="px-4 bg-opacity-20 min-w-[125px] h-10"
        >
          <p
            class="text-m-sm sm:text-t-sm md:text-d-sm text-grey text-left font-regular"
          >
            children
          </p>
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;

exports[`Test TableCell Variants variant="body 1`] = `
<DocumentFragment>
  <table
    class="w-full"
  >
    <tbody
      class="table-row-group"
    >
      <tr
        class="border-b border-grey-100 border-opacity-20 hover:bg-opacity-75 bg-white border-grey-light"
      >
        <td
          class="px-4 bg-opacity-20 min-w-[125px] h-10"
        >
          <p
            class="text-m-sm sm:text-t-sm md:text-d-sm text-grey text-left font-regular"
          >
            children
          </p>
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;

exports[`Test TableCell Variants variant="footer 1`] = `
<DocumentFragment>
  <table
    class="w-full"
  >
    <tbody
      class="table-row-group"
    >
      <tr
        class="border-b border-grey-100 border-opacity-20 hover:bg-opacity-75 bg-white border-grey-light"
      >
        <td
          class="px-4 bg-opacity-20 min-w-[125px] h-10"
        >
          <p
            class="text-m-sm sm:text-t-sm md:text-d-sm text-grey text-left font-regular"
          >
            children
          </p>
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;

exports[`Test TableCell Variants variant="head 1`] = `
<DocumentFragment>
  <table
    class="w-full"
  >
    <tbody
      class="table-row-group"
    >
      <tr
        class="border-b border-grey-100 border-opacity-20 hover:bg-opacity-75 bg-white border-grey-light"
      >
        <th
          class="px-4 bg-opacity-20 min-w-[125px] h-10"
        >
          <p
            class="text-m-sm sm:text-t-sm md:text-d-sm text-grey text-left font-regular"
          >
            children
          </p>
        </th>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;
